LibRangeCheck-2.0/LibRangeCheck-2.0.lua

```diff
@@ -1030,29 +1047,46 @@ function lib:GetRange(unit, checkVisible, noItems)
         return nil
     end
 
+    local currentTime = GetTime()
+    local guid = UnitGUID(unit)
+    local cacheItem = rangeCache[guid]
+    if cacheItem and cacheItem.updateTime + self.getRangeThrottle > currentTime then
+        return cacheItem.minRange, cacheItem.maxRange
+    end
+
+    local result = cacheItem or {}
+
     local canAssist = UnitCanAssist("player", unit)
     if UnitIsDeadOrGhost(unit) then
         if canAssist then
-            return getRange(unit, self.resRC)
+            result.minRange, result.maxRange = getRange(unit, self.resRC)
         else
-            return getRange(unit, self.miscRC)
+            result.minRange, result.maxRange = getRange(unit, self.miscRC)
         end
     end
 
     if UnitCanAttack("player", unit) then
-        return getRange(unit, noItems and self.harmNoItemsRC or self.harmRC)
+        result.minRange, result.maxRange = getRange(unit, noItems and self.harmNoItemsRC or self.harmRC)
     elseif UnitIsUnit("pet", unit) then
-        local minRange, maxRange = getRange(unit, noItems and self.friendNoItemsRC or self.friendRC)
-        if minRange or maxRange then
-            return minRange, maxRange
-        else
-            return getRange(unit, self.petRC)
+        result.minRange, result.maxRange = getRange(unit, noItems and self.friendNoItemsRC or self.friendRC)
+        if not result.minRange and not result.maxRange then
+            result.minRange, result.maxRange = getRange(unit, self.petRC)
         end
     elseif canAssist then
-        return getRange(unit, noItems and self.friendNoItemsRC or self.friendRC)
+        result.minRange, result.maxRange = getRange(unit, noItems and self.friendNoItemsRC or self.friendRC)
     else
-        return getRange(unit, self.miscRC)
+        result.minRange, result.maxRange = getRange(unit, self.miscRC)
     end
+
+    result.updateTime = currentTime
+    rangeCache[guid] = result
+    return result.minRange, result.maxRange
+end
+
+--- Sets the time the cache entries for range checks with GetRange are valid
+-- @param timespan the time in seconds (default is 0.1)
+function lib:SetGetRangeThrottle(timespan)
+    self.getRangeThrottle = timespan or 0.1
 end
```

Comment:
1. I don't like the idea that an addon can change the throttle for all addons
2. A table field starting with `get.....` should be a function returning a value, i'd call it self.rangeThrottle
3. `SetGet`?

Reply:
1. I see three possible solutions.
    a. Use a fixed value.
    b. Set it per caller and provide some kind of caller context to the `GetRange` method.
    c. Provide a maximum age to the `GetRange` method with a default (my favorite).
2. You are right, I'll change that. I only named it like that because its a throttle for the `getRange` method.
3. Yea that's a bit stupid ... because its a setter to the `getRangeThrottle` field. But this setter is maybe obsolete anyway because of point 1.